lib/task_helpers/metrics.rb

```diff
@@ -0,0 +1,41 @@
+module TaskHelpers
+  class Metrics
+    class << self
+      def per_tenant
+        column_names = %w[
+          tenant
+          portfolio_count
+          product_count
+          portfolio_share_count
+        ].freeze
+        data = [column_names.collect(&:titleize)]
+
+        Tenant.order(:external_tenant).each do |current_tenant|
+          ActsAsTenant.with_tenant(current_tenant) do
+            data << column_names.collect { |column_name| send(column_name) }
```

Comment:
@pkomanek We auto-generate the tenant object for any user that hits the API and the report ends up with a log of tenants rows that just have zero data points

I'd like to add a check at the top of the `with_tenant` block to skip a tenant if there are no Portfolios or Orders so we can reduce the amount of "noise" in the report.

We could put this check into a method so if could be tested separately.
```ruby
def tenant_empty?
  Order.count.zero? && Portfolio.count.zero?
end
```
And the block condition would be:
```ruby
next if tenant_empty?
```